### PR TITLE
Manual.md: improve automatic depends section, adjust sections hierarchy

### PR DESCRIPTION
**Pull request by Chocimier**

### CODE REVIEW

File: Manual.md [hunk 1]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
 
+<a id="deps_runtime"></a>
 Lastly, a package may require certain dependencies at runtime, without which it
 is unusable. These dependencies, when they aren't detected automatically by
-XBPS, should be listed in `depends`. This is mostly relevant for Perl and Python
-modules and other programs that use `dlopen(3)` instead of dynamically linking.
+XBPS, should be listed in `depends`.
+
+Libraries linked by ELF objects are detected automatically by `xbps-src`, hence they
+must not be specified in templates via `$depends`. Variable should list:
```
Comment (mhmdanas) [inline Manual.md:797]:
```suggestion
must not be specified in templates via `depends`. Variable should list:
```

File: Manual.md [hunk 2]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
+
+- executables called as separate processes.
+- ELF objects using dlopen(3).
+- non ELF objects, i.e perl/python/ruby/etc modules.
```
Comment (mhmdanas) [inline Manual.md:801]:
```suggestion
- non-ELF objects, i.e perl/python/ruby/etc modules.
```

File: Manual.md [hunk 3]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
+- data files.
+- overriding the minimal version specified in the `shlibs` file.
```
Comment (mhmdanas) [inline Manual.md:803]:
```suggestion
- overriding the minimal version specified in the `common/shlibs` file.
```

File: Manual.md [hunk 4]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
+
+The runtime dependencies for ELF objects are detected by checking which SONAMEs
+they require and then the SONAMEs are mapped to a binary package name with a minimal
+required version. The `shlibs` file in the `void-packages/common` directory
+sets up the `<SONAME> <pkgname>>=<version>` mappings.
```
Comment (mhmdanas) [inline Manual.md:808]:
```suggestion
required version. The `common/shlibs` file sets up the
`<SONAME> <pkgname>>=<version>` mappings.
```

File: Manual.md [hunk 5]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
+
+For example the `foo-1.0_1` package provides the `libfoo.so.1` SONAME and
+software requiring this library will link to `libfoo`; the resulting binary
+package will have a run-time dependency to `foo>=1.0_1` package as specified in
+`common/shlibs`:
```
Comment (mhmdanas) [inline Manual.md:813]:
```suggestion
For example the `foo-1.0_1` package provides the `libfoo.so.1` SONAME and
software requiring this library will link to `libfoo.so.1`; the resulting binary
package will have a run-time dependency on `foo>=1.0_1` package as specified in
`common/shlibs`:
```

File: Manual.md [hunk 6]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
+...
+libfoo.so.1 foo-1.0_1
+...
+```
+
+- The first field specifies the SONAME.
+- The second field specified the package name and minimal version required.
+- A third optional field (usually set to `ignore`) can be used to skip checks in soname bumps.
+
+Dependencies declared via `${depends}` are not installed to the master directory, rather are
```
Comment (mhmdanas) [inline Manual.md:826]:
```suggestion
Dependencies declared via `depends` are not installed to the master directory, rather are
```

File: Manual.md [hunk 7]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
+As a special case, `virtual` dependencies may be specified as runtime dependencies in the
+`${depends}` template variable. Several different packages can provide common functionality by
+declaring a virtual name and version in the `${provides}` template variable (e.g.,
+`provides="vpkg-0.1_1"`). Packages that rely on the common functionality without concern for the
+specific provider can declare a dependency on the virtual package name with the prefix `virtual?`
+(e.g., `depends="virtual?vpkg-0.1_1"`). When a package is built by `xbps-src`, providers for any
+virtual packages will be confirmed to exist and will be built if necessary. A map from virtual
+packages to their default providers is defined in `etc/defaults.virtual`. Individual mappings can be
+overridden by local preferences in `etc/virtual`. Comments in `etc/defaults.virtual` provide more
+information on this map.
```
Comment (mhmdanas) [inline Manual.md:839]:
```suggestion
As a special case, `virtual` dependencies may be specified as runtime dependencies in the
`depends` template variable. Several different packages can provide common functionality by
declaring a virtual name and version in the `provides` template variable (e.g.
`provides="vpkg-0.1_1"`). Packages that rely on the common functionality without concern for the
specific provider can declare a dependency on the virtual package name with the prefix `virtual?`
(e.g., `depends="virtual?vpkg-0.1_1"`). When a package is built by `xbps-src`, providers for any
virtual packages will be confirmed to exist and will be built if necessary. A map from virtual
packages to their default providers is defined in `etc/defaults.virtual`. Individual mappings can be
overridden by local preferences in `etc/virtual`. Comments in `etc/defaults.virtual` provide more
information on this map.
```

**Review by mhmdanas — COMMENTED** [review]
I know much of the text was simply moved from one place to another, but here are my nits anyway.

File: Manual.md (hunk 2 above)
Comment (Chocimier) [reply Manual.md]: Thanks for remarks.

Here also abstracted over object format.

File: Manual.md [hunk 8]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
 
+<a id="deps_runtime"></a>
 Lastly, a package may require certain dependencies at runtime, without which it
 is unusable. These dependencies, when they aren't detected automatically by
-XBPS, should be listed in `depends`. This is mostly relevant for Perl and Python
-modules and other programs that use `dlopen(3)` instead of dynamically linking.
+XBPS, should be listed in `depends`.
+
+Libraries linked by ELF objects are detected automatically by `xbps-src`, hence they
+must not be specified in templates via `depends`. Variable should list:
```
Comment (classabbyamp) [inline Manual.md:797]:
```suggestion
must not be specified in templates via `depends`. This variable should list:
```

File: Manual.md [hunk 9]
```diff
@@ -787,28 +788,73 @@ should be listed in `checkdepends` and will be installed as if they were part of
...
+...
+libfoo.so.1 foo-1.0_1
+...
+```
+
+- The first field specifies the SONAME.
+- The second field specified the package name and minimal version required.
+- A third optional field (usually set to `ignore`) can be used to skip checks in soname bumps.
+
+Dependencies declared via `depends` are not installed to the master directory, rather are
```
Comment (Bnyro) [inline Manual.md:826]:
```suggestion
Dependencies declared via `depends` are not installed to the master directory, rather they are
```

**Pull request merged**